### ISSUE TITLE
Replace ShareThis Social buttons with "react-share" and optimize for screen readers

### ISSUE THREAD

**Issue by blnkt** [lsst-epo/rubin-obs-client#52]
Related to #58 

## Expected Behavior
Developers should be able to directly edit and dictate markup for Share buttons; Share lib should just only in charge of functionality of the links/buttons, not their appearance.  Especially should be able to use semantic html (e.g. button or link) and be able to provide appropriate alt  or screen reader only text.

## Actual Behavior
ShareThis locks the dev into a non-semantic html pattern that provides few options for customization beyond style overrides.

<img width="940" alt="ddc33c08378944a3f0d25f417d300d08941fd9ab-1641212485" src="https://user-images.githubusercontent.com/8166060/152439373-40ab156b-8d8b-4c9f-a245-7a0cfb417f44.png">

## Suggested improvement
@blnkt used a react package called [`react-share`](https://www.npmjs.com/package/react-share) in the skyviewer project which seems to work quite well, requires no additional service account, and is extremely customizable.  Checkout [this set of ShareButtons in the skyviewer repo](https://github.com/lsst-epo/skyviewer-client/tree/master/components/ShareButtons) for a better indication of implementation.  If CIC has experience with another similar package EPO is happy to defer to their expert recommendation.

## CIC Estimate
1 point
